bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,606 @@
+
+                        if (LOG.isDebugEnabled()) {
+                            LOG.debug("Session has been saved and refreshed.");
+                        }
+                    }
+                }
+            } else if (json.get(key) instanceof JSONArray) {
+                try {
+                    if (PROTECTED_PROPERTIES.contains(key)) {
+                        // Skipping due to the property being unmodifiable.
+                        continue;
+                    } else if (JcrConstants.JCR_MIXINTYPES.equals(key)) {
+                        setMixinsProperty(json, key, parentNode);
+                    } else if (NameConstants.PN_TAGS.equals(key)) {
+                        setTagsProperty(json, key, parentNode);
+                    } else {
+                        setArrayProperty(json, key, parentNode);
+                    }
+                } catch (RepositoryException re) {
+                    LOG.warn("Repository exception thrown. Skipping {} array property for node '{}'.", key, parentNode.getPath());
+                }
+            } else {
+                try {
+                    Object value = json.get(key);
+
+                    if (":".concat(JcrConstants.JCR_DATA).equals(key)) {
+                        setJcrDataProperty(parentNode, json.getString(JcrConstants.JCR_LASTMODIFIED));
+                    } else if (key.startsWith(":")) {
+                        // Skip binary properties, since they do not come across in JSON
+                        continue;
+                    } else if (parentNode.hasProperty(key) && parentNode.getProperty(key).getString().equals(value)) {
+                        // Skipping due to the property already existing and being equal
+                        continue;
+                    } else if (PROTECTED_PROPERTIES.contains(key)) {
+                        // Skipping due to the property being unmodifiable.
+                        continue;
+                    } else {
+                        setProperty(value, key, parentNode);
+                    }
+                } catch (RepositoryException re) {
+                    LOG.warn("Repository exception thrown. Skipping '{}' single property for node '{}'.", key, parentNode.getPath());
+                }
+            }
+        }
+    }
+
+    /**
+     * Set mixins property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setMixinsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray mixins = (JSONArray) json.get(key);
+        for (int i = 0; i < mixins.length(); i++) {
+            node.addMixin(mixins.getString(i));
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Mixin '{}' added for node '{}'.", mixins.getString(i), node.getPath());
```

Comment:
I recommend putting this before adding the mixin - that way if it fails (exception) we know what mixin was tried last.